test/functional/p2p_sendtxrcncl.py

```diff
@@ -217,7 +217,7 @@ def run_test(self):
 
         self.log.info('SENDTXRCNCL without WTXIDRELAY is ignored (recon state is erased after VERACK)')
         peer = self.nodes[0].add_p2p_connection(PeerNoVerack(wtxidrelay=False), send_version=True, wait_for_verack=False)
-        with self.nodes[0].assert_debug_log(['Forget txreconciliation state of peer']):
+        with self.nodes[0].assert_debug_log(['Register peer', 'Forget txreconciliation state of peer']):
```

Comment:
Approach nack on using logs to test behavior. Is this covered in unit tests?

Reply:
> Is this covered in unit tests?

No, it doesn't cover any p2p iteration.

Reply:
Tbh, most part of the functional test relies on checking logs which I agree it's a bad way to test behavior. However, since we're just registering and forgetting peers, there is no other way to test stuff on the functional side, except disconnections.

I'll close this for now and suggest some improvements for this test according to the Erlay PRs.